src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs

```diff
@@ -47,6 +48,77 @@ protected static class ArtifactEventProperties
         }
     }
 
+    // Caller: PublishPipelineArtifact task
+    // Can be invoked from a build run or a release run should a build be set as the artifact. 
+    public class PublishPipelineArtifactTaskV1 : PipelineArtifactTaskPluginBaseV1
+    {
+        public override Guid Id => PipelineArtifactPluginConstants.PublishPipelineArtifactTaskId;
+        protected override string TargetPath => "path";
+
+        public static readonly Regex jobIdentifierRgx = new Regex("[^a-zA-Z0-9 - .]", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        protected override async Task ProcessCommandInternalAsync(
+            AgentTaskPluginExecutionContext context, 
+            CancellationToken token)
+        {           
+            string artifactName = context.GetInput(ArtifactEventProperties.ArtifactName, required: false);
+            string targetPath = context.GetInput(TargetPath, required: true);
+            string defaultWorkingDirectory = context.Variables.GetValueOrDefault("system.defaultworkingdirectory").Value;
+
+            targetPath = Path.IsPathFullyQualified(targetPath) ? targetPath : Path.GetFullPath(Path.Combine(defaultWorkingDirectory, targetPath));
+
+            string hostType = context.Variables.GetValueOrDefault("system.hosttype")?.Value; 
+            if (!string.Equals(hostType, "Build", StringComparison.OrdinalIgnoreCase)) {
+                throw new InvalidOperationException(
+                    StringUtil.Loc("CannotUploadFromCurrentEnvironment", hostType ?? string.Empty)); 
+            }
+
+            if (String.IsNullOrWhiteSpace(artifactName))
+            {
+                string jobIdentifier = context.Variables.GetValueOrDefault("system.jobIdentifier").Value;
+                var normalizedJobIdentifier = NormalizeJobIdentifier(jobIdentifier);
+                artifactName = normalizedJobIdentifier;
+            }
+
+            if(!PipelineArtifactPathHelper.IsValidPath(artifactName)) {
+                throw new ArgumentException(StringUtil.Loc("ArtifactNameIsNotValid"));
+            }
+
+            // Project ID
+            Guid projectId = new Guid(context.Variables.GetValueOrDefault(BuildVariables.TeamProjectId)?.Value ?? Guid.Empty.ToString());
+            ArgUtil.NotEmpty(projectId, nameof(projectId));
+
+            // Build ID
+            string buildIdStr = context.Variables.GetValueOrDefault(BuildVariables.BuildId)?.Value ?? string.Empty;
```

Comment:
>?? string.Empty; [](start = 99, length = 16)

Do not default if the field is required, always fail right away with GetValue

Reply:
We don't want to change this behavior in V1 task? It's the same as the V0.

Reply:
see above